### DISCUSSION TITLE
Pass options to pino logger

### DISCUSSION THREAD

**Discussion by rishi-afk** [payloadcms/payload#418] [Feature Requests & Ideas]
Looks like Payload is making use of Pino for logging. It would be great to see if we could pass an optional configuration to the Payload config to setup Pino for a more standard logging setup to log data along with the request context and also disabling pretty json print. 
This would help a lot as I am using Payload as a microservice and would benefit from a more standard logging setup. Any feedbacks are welcome. Thank you.

**Comment by denolfe** [comment]
Hey @rishi-afk, this is something we can look into. I don't think it would be too difficult to add some logger override options to the payload config. Thoughts on that?

**Reply by rishi-afk** [reply]
Hi @denolfe, yes having an option to configure pino from the payload config would be perfect and doesn't look that difficult. This feature would help a lot.

**Reply by zaunermax** [reply]
Hi, we have the same issue where we use file logging and there is no way to alter logging options :raised_hands: Altering the logging configuration would help us as well :+1:

**Reply by rishi-afk** [reply]
Hello @denolfe, any updates regarding this?

**Reply by denolfe** [reply]
I've done some initial investigation on this. We do some logging _before_ loading the Payload config, so I'm mulling over the best way to go for that. No ETA, but this is definitely considered planned work  in the next few weeks.

**Reply by zaunermax** [reply]
@denolfe I remember that being a problem in the nestjs framework as well. So they then introduced a parameter which was smth like `bufferLogs` which lets you decide if all logs shoud adhere to a custom config which as loaded later on in the bootstrap process or if nest just logged the bootstrap logs with its own config and then continued with the custom config once it was loaded. Maybe this is useful input :raised_hands:

**Reply by jmikrut** [reply]
Hey @rishi-afk — this has now been deployed as of `0.15.4` and documentation can be found [here](https://payloadcms.com/docs/getting-started/installation#server)!

**Reply by rishi-afk** [reply]
@denolfe @jmikrut This looks great. Finally we can move into production. Payload is just getting better and better 🙌

**Reply by DanRibbens** [reply]
Just a heads up—we had a bug in yesterdays release that is now fixed in `0.15.5`.
